x-pack/plugins/task_manager/server/task.ts

```diff
@@ -83,58 +85,62 @@ export interface FailedTaskResult {
   status: TaskStatus.Failed;
 }
 
-export const validateRunResult = Joi.object({
-  runAt: Joi.date().optional(),
-  schedule: Joi.object().optional(),
-  error: Joi.object().optional(),
-  state: Joi.object().optional(),
-}).optional();
-
 export type RunFunction = () => Promise<RunResult | undefined | void>;
-
 export type CancelFunction = () => Promise<RunResult | undefined | void>;
-
 export interface CancellableTask {
   run: RunFunction;
   cancel?: CancelFunction;
 }
 
 export type TaskRunCreatorFunction = (context: RunContext) => CancellableTask;
 
+export const taskDefinitionSchema = schema.object(
+  {
+    /**
+     * A unique identifier for the type of task being defined.
+     */
+    type: schema.string(),
+    /**
+     * A brief, human-friendly title for this task.
+     */
+    title: schema.maybe(schema.string()),
+    /**
+     * An optional more detailed description of what this task does.
+     */
+    description: schema.maybe(schema.string()),
+    /**
+     * How long, in minutes or seconds, the system should wait for the task to complete
+     * before it is considered to be timed out. (e.g. '5m', the default). If
+     * the task takes longer than this, Kibana will send it a kill command and
+     * the task will be re-attempted.
+     */
+    timeout: schema.string({
+      defaultValue: '5m',
+    }),
+    /**
+     * Up to how many times the task should retry when it fails to run. This will
+     * default to the global variable.
+     */
+    maxAttempts: schema.maybe(
+      schema.number({
+        min: 1,
+      })
+    ),
+  },
+  {
+    validate({ timeout }) {
+      if (!isInterval(timeout) || isErr(tryAsResult(() => parseIntervalAsMillisecond(timeout)))) {
+        return `Invalid timeout "${timeout}". Timeout must be of the form "{number}{cadance}". Example: 5m.`;
```

Comment:
nit: if the timeout was `1.5m` and I received this error message, I wouldn't immediately know what was wrong because 1.5 is technically a number. Maybe it would be helpful to say something like `Invalid timeout "${timeout}". Timeout must be of the form "{number}{cadance}" where number is an integer. Example: 5m.`

Reply:
Fair 👍